src/main/java/org/elasticsearch/index/query/functionscore/fieldvaluefactor/FieldValueFactorFunctionParser.java

```diff
@@ -70,6 +70,8 @@ public ScoreFunction parse(QueryParseContext parseContext, XContentParser parser
                 } else {
                     throw new QueryParsingException(parseContext.index(), NAMES[0] + " query does not support [" + currentFieldName + "]");
                 }
+            } else if("factor".equals(currentFieldName) && token == XContentParser.Token.START_ARRAY) {
```

Comment:
Can you add a check for `XContentParser.Token.START_OBJECT` as well here, since it can't be an object either?